subliminal/providers/legendastv.py

```diff
@@ -0,0 +1,424 @@
+# -*- coding: utf-8 -*-
+import os
+import logging
+import re
+import json
+from zipfile import ZipFile, is_zipfile
+from rarfile import RarFile, is_rarfile
+from tempfile import NamedTemporaryFile
+
+from babelfish import Language, language_converters
+from requests import Session
+
+from . import ParserBeautifulSoup, Provider, get_version
+from .. import __version__
+from ..cache import SHOW_EXPIRATION_TIME, region
+from ..exceptions import AuthenticationError, ConfigurationError
+from subliminal.exceptions import ProviderError
+from ..subtitle import Subtitle, fix_line_ending, guess_matches, compute_score
+from ..video import Episode, Movie, SUBTITLE_EXTENSIONS
+from guessit import guess_file_info
+
+TIMEOUT = 10
+
+logger = logging.getLogger(__name__)
+
+server_url = 'http://legendas.tv'
+auth_error_msg_re = re.compile(u'.*Usuário ou senha inválidos.*')
+season_re = re.compile('.*? - (\d{1,2}).*?emporada')
+rating_info_re = re.compile('(\d*) downloads, nota (\d{0,2})')
+imdb_re = re.compile('t{0,2}(\d+)')
+subtitle_href_re = re.compile('/download/(\w+)/.+')
+
+mapping = dict(
+    id='id_filme',
+    name='dsc_nome',
+    name_br='dsc_nome_br',
+    year='dsc_data_lancamento',
+    type='tipo',
+    season='temporada',
+    imdb_id='id_imdb'
+)
+
+typemap = {
+    'M': 'movie',
+    'S': 'episode',
+    'C': 'episode'  # Considering C as episode. Probably C stands for Collections
+}
+
+
+def has_class(tag, expected_class):
+    while tag:
+        if not tag:
+            return False
+        if tag.has_attr('class') and expected_class in tag['class']:
+            return True
+        tag = tag.parent
+    return False
```

Comment:
I'm not sure what this does but I believe there's a BeautifulSoup function for that or you need to change the way you do it.